### DISCUSSION TITLE
Calling REST API returning data

### DISCUSSION THREAD

**Discussion by offpiste-reese** [Budibase/budibase#5309] [Help]
I'm looking at Budibase for building an app that pulls data from Lime Survey, aggregates it and uses the aggregated data to create a PDF.  I'm a little confused on how to get data out of the return value from REST calls in Budibase.  I've attached a screen shot of a POC I created using Bubble (another tool we're looking at) for reference.  Basically, I want to have a form with user id and password.  Click login, it calls the Lime API and logs the user into Lime and puts the session id in a text box.  Click the surveys button, and the list of surveys available to this user is pulled via the Lime API and displayed in a list box.  Select a survey, click a button and the Lime API is called to display the list of survey takers  is displayed in another list box.  This is illustrated in the attached screen shot.  

Basically, I'm confused on how to get the data returned by REST calls displayed in various controls.  I've looked at the documentation, but perhaps I'm not looking the correct place.  Any insight would be appreciated. 
![BubbleFFPOC](https://user-images.githubusercontent.com/103227709/162342249-17843c88-500d-4f3a-9a40-16a2bec51b7c.png)

**Comment by melohagan** [comment]
Hi @offpiste-reese 

You could using action referencing (https://docs.budibase.com/docs/referencing) and App State (https://docs.budibase.com/docs/app-state) 

So your log-in button would have two actions:
 - Execute Query (to get the session id)
 - Update State (to put the session id from the query into state)

Then you could access the session id anywhere on the page with the binding: `{{ State.<session-id-name-here> }}`
